nativelib/src/main/resources/gc/immix/datastructures/Bytemap.h

```diff
@@ -0,0 +1,114 @@
+#ifndef IMMIX_BYTEMAP_H
+#define IMMIX_BYTEMAP_H
+
+#include <stddef.h>
+#include <stdbool.h>
+#include <string.h>
+#include "../GCTypes.h"
+#include "../Constants.h"
+#include "../Log.h"
+
+typedef struct {
+    word_t *firstAddress;
+    size_t size;
+    ubyte_t *end;
+    ubyte_t data[0];
+} Bytemap;
```

Comment:
I have an impression that this module has two conceptual things merged in one: byte mapping and object metadata. When you access a bytemap you should get a pointer to object metadata back, and getters/setters to modify the header should go into its own module. Object metadata might evolve to store more info, so it's best to keep it separate conceptually.